### ISSUE TITLE
The server does not start after update to fastify v4.28.1

### ISSUE THREAD

**Issue by pasha-vuiko** [Vanilla-IceCream/fastify-uws#1]
The server does not start after update to fastify v4.28.1, previously worked fine. The server just being freezed on call of .listen()

**Comment by Shyam-Chen** [comment]
I cannot reproduce your issue; I am using Node.js v20.

Ref: https://github.com/Shyam-Chen/Fastify-Starter/blob/main/app/package.json#L33

--- 

Updated:

Downgraded to version 4.28.0

**Comment by pasha-vuiko** [comment]
Hmm, interesting, let me check again

**Comment by pasha-vuiko** [comment]
Still have the issue, previously tried with nestjs-platform-fastify, now tried with pure fastify:

```javascript
import fastify from "fastify";
import { serverFactory } from "fastify-uws";

const app = fastify({
  serverFactory,
});

app.get("/", async (request, reply) => {
  const hello = 'hello';

  return { hello: hello };
});

app
  .listen({
    host: "0.0.0.0",
    port: 3000,
  })
  .then((address) => {
    console.log(`server listening on ${address}`);
  });
```
the issue is still the same, the server just not starting.

I'm using MacOS 14.5 and Node.js v20.15.0

**Comment by Shyam-Chen** [comment]
`listen` works, but `.then()` cannot be executed.

> The changes affect `listen` in version 4.28.1:
> https://github.com/fastify/fastify/pull/5523/files#diff-1b48bb429de959672dadab9c87c215e19177dd70e3bd932d131ce74294c44cd5

--- 

Upstream Issue Tracking: https://github.com/geut/fastify-uws/issues/6

**Comment by Shyam-Chen** [comment]
Released in `fastify-uws` version 0.8, incorporating some upstream code.

---

```sh
 ~/Documents/Fastify-Starter/ [main] pnpm dev

> @ dev /Users/shyam.chen/Documents/Fastify-Starter
> pnpm -C app dev


> @ dev /Users/shyam.chen/Documents/Fastify-Starter/app
> vite-node -w src/main.ts

2024-07-16 17:19:54.367+0800 - info - Server listening at http://127.0.0.1:3000
```
